pkg/eval/vals/equal.go

```diff
@@ -22,7 +25,17 @@ func Equal(x, y interface{}) bool {
 	case bool:
 		return x == y
 	case float64:
-		return x == y
+		if math.IsNaN(x) && math.IsNaN(y.(float64)) {
```

Comment:
I don't like the idea of making floating point equality smart; best to remove this change from the PR.

Reply:
Fair enough. This was mostly a Go learning experience and a "what if...". But as the subsequent comment explains this was also an expedient way to test that NaN is output in unit tests. I'll revert this and give more thought to the unit test case.